cmd/kv-tcp/main.go

```diff
@@ -81,22 +86,46 @@ func main() {
 	store = serializable.NewTwoPhaseLockStore(store)
 	transactor := transactors.New(store, writer)
 
-	server{store, transactor}.serve(ln.(*net.TCPListener))
+	s := server{ln.(*net.TCPListener), store, transactor, make(chan bool, 1)}
+
+	stopSignal := make(chan os.Signal, 1)
+	signal.Notify(stopSignal, syscall.SIGTERM, syscall.SIGINT)
+	go func() {
+		sig := <-stopSignal
+		logrus.Infof("Received %v signal\n", sig)
+		s.stop()
+		os.Exit(0)
+	}()
+
+	s.serve()
 }
 
 type server struct {
+	l          net.Listener
 	store      stores.Store
 	transactor transactors.Transactor
+	quit       chan bool
```

Comment:
Since the Boolean value appears to be unused, consider making this a `chan struct{}` and signaling it with `close(quit)`. This avoids an allocation, and makes it more obvious that this channel is just a one-time trigger.